src/main/scala/mirroring/services/writer/DeltaService.scala

```diff
@@ -16,13 +16,18 @@
 
 package mirroring.services.writer
 
+import io.delta.tables.DeltaTable
 import mirroring.builders.FilterBuilder
+import mirroring.services.SparkService.spark
 import org.apache.spark.sql.{DataFrame, DataFrameWriter, Row}
 import wvlet.log.LogSupport
 
 class DeltaService(context: WriterContext) extends LogSupport {
 
   def write(data: DataFrame): Unit = {
+    if (DeltaTable.isDeltaTable(spark, context.path)) {
```

Comment:
is this check needed? AFAIK, if mergeSchema is false, [schema enforcement](https://delta.io/blog/2022-11-16-delta-lake-schema-enforcement/) is in place.

Reply:
Yes, it is needed. If the table is mirrored at the first time, one doesn't have a delta table, so `AnalysisException: Path does not exist` is raised. Another way is `try .. catch`  block catching `case e: org.apache.spark.sql.AnalysisException if e.getMessage().startsWith("...` but it seems to be less preferred.